src/amo/sagas/addons.js

```diff
@@ -42,7 +44,49 @@ export function* fetchAddon({
     yield put(loadAddon({ addon, slug }));
   } catch (error) {
     log.warn(`Failed to load add-on with slug ${slug}: ${error}`);
-    yield put(errorHandler.createErrorAction(error));
+
+    if ([401, 403].includes(error.response?.status) && assumeUnlisted) {
```

Comment:
Not new to your patch, so maybe we should deal with this in a follow-up, but as mentioned above this should also apply to non-public even if they were listed (deleted, rejected, disabled etc - doesn't matter, it should still be possible to file a report against them).

Which in turn means that... at least if it looks like a guid, we should handle deleted too and include `404` here and never worry about returning a `<NotFound>` for the component above...

Additionally since those reports can only come from Firefox users (no AMO link to follow, unless incredibly unfortunate timing), we could rely on `mozAddonManager` to tell us if the add-on is installed to "validate" that the `guid` exists on the user's Firefox before proceeding on the client-side, if we're worried about basic spamming?

Reply:
An example on dev is `{34f676d6-32ef-4833-adff-f84e992bb7c5}` which was listed before getting deleted. (Note: if you're logging in with the right permissions you'd be able to load it anyway. Make sure to be logged out when testing)

Reply:
I filed